sunkit_instruments/rhessi/rhessi.py

```diff
@@ -350,3 +351,60 @@ def _build_energy_bands(label, bands):
     unit = matched.group('UNIT').strip()
 
     return [f'{band} {unit}' for band in bands]
+
+
+def hsi_fits2map(image_datacube):
+    """
+    Extracts single map images from a RHESSI flare image datacube.
+    Currently assumes input to be 4D.
+
+    Parameters
+    ----------
+    image_datacube : `str`
+        Path or URL to image datacube .fits
+
+    Returns
+    -------
+    `dict` of `list` of `sunpy.map.Map`
+        Each energy band has a list of maps where the index of the lists represent the time step
+    """
+    # import sunpy.map in here so that net and timeseries don't end up importing map
+    from sunpy.map import Map
+
+    f = sunpy.io.read_file(image_datacube)
+    header = f[0].header
+
+    # make sure datacube is a RHESSI cube
+    if header['INSTRUME'] != "RHESSI":
+        raise ValueError(f"Expected a RHESSI datacube, got: {header['INSTRUME']}")
+
+    # remove those (non-standard) headers to avoid user warnings (they are 0 anyway)
+    del header["CROTACN1"]
+    del header["CROTACN2"]
+    del header["CROTA"]
+
+    d_min = {}
+    d_max = {}
+    e_ax = f[1].data[0]["ENERGY_AXIS"].reshape((-1, 2))  # reshape energy axis to be 2D
+    t_ax = f[1].data[0]["TIME_AXIS"].reshape((-1, 2))  # reshape time axis to be 2D
+    data = f[0].data.reshape(tuple([1] * (4 - len(f[0].data.shape))) + f[0].data.shape)  # reshape data to be 4D
+    for e in range(e_ax.shape[0]):
+        d_min[e] = 1e10
+        d_max[e] = -1e10
+        for t in range(t_ax.shape[0]):
+            d_min[e] = min(d_min[e], data[t][e].min())
+            d_max[e] = max(d_max[e], data[t][e].max())
+
+    maps = {}  # result dictionary
+    for e in range(e_ax.shape[0]):
+        header["ENERGY_L"] = e_ax[e][0]
+        header["ENERGY_H"] = e_ax[e][1]
+        header["DATAMIN"] = d_min[e]
+        header["DATAMAX"] = d_max[e]
+        key = f"{int(header['ENERGY_L'])}-{int(header['ENERGY_H'])} keV"
+        maps[key] = []
+        for t in range(t_ax.shape[0]):
+            header["DATE_OBS"] = parse_time(t_ax[t][0], format='utime').to_value('isot')
+            header["DATE_END"] = parse_time(t_ax[t][1], format='utime').to_value('isot')
+            maps[key].append(Map(data[t][e], header))  # extract image Map
```

Comment:
```suggestion
            map_list.append(Map(data[t][e], header))  # extract image Map
        if len(map_list)==1:
            maps[key] = Map(map_list)
        else:
            maps[key] = Map(map_list, sequence=True)
```

Reply:
Thanks for the suggestion!
Like @DanRyanIrish I prefer it was always was a sequence...

in practice, looping changes from
```python
for energy_band in maps.keys():
    for m in maps[energy_band]:
        #do stuff
```
to
```python
from sunpy.map import GenericMap
for energy_band in maps.keys():
    for m in [maps[energy_band]] if isinstance(maps[energy_band], GenericMap) else maps[energy_band]:
        #do stuff
```

which is a bit unnecessarily complicated for my taste, but manageable. I can totally live with it too, but, I don't really see the advantage tbh